### ISSUE TITLE
Record.getValue(Field) returns wrong value if ambiguous column names are contained in the record, and the schema name is not present in the argument

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#4471]
When fetching data from the database that contains ambiguous column names, e.g.

```sql
select b.id, a.id 
from t_author a 
join t_book b 
on a.id = b.author_id 
order by b.id, a.id
```

... resulting in

```
+----+----+
|  id|  id|
+----+----+
|   1|   1|
|   2|   1|
|   3|   2|
|   4|   2|
+----+----+
```

Then the `Record.getValue(Field)` or `Result.getValue(Field)` method doesn't work correctly, if the schema name isn't present in both:

- The record's field
- The argument's field

E.g., since #4283, the schema name is fetched from those JDBC drivers that support it, but not from PostgreSQL. When getting the value via `record.getValue(field(name("public", "t_author", "id")))` in PostgreSQL, the value is wrong (that of `t_book.id`).

Conversely, when using generated tables / columns, the schema is always present. But when getting the value via `record.getValue(field(name("t_author", "id")))`, the value is again wrong.

----
See also: #4283

**Comment by lukaseder** [comment]
This won't be merged as it depends partially on #4283

**Comment by fedotxxl** [comment]
is it fixed? I use `3.7.1` and still have this issue. How to correctly map in this case joined tables to java object? Don't this issue makes joining in PostgreSQL pretty dangerous and useless?

**Comment by lukaseder** [comment]
There have been several similar reported issues in the past. Not all of the issue reports were correct, thus they may not have been "fixed" as you expect it. For instance, the behaviour may be perceived as being subtly different for generated code than for plain SQL API usage. This difference may or may not be a bug.

Of course, there are two questions here:

1. What does your code look like?
2. Why does it make you conclude that joining (in general) is dangerous?
